sunpy/coordinates/ephemeris.py

```diff
@@ -285,10 +285,12 @@ def get_horizons_coord(body, time='now', id_type='majorbody', *, include_velocit
                      epochs=array_time.tdb.jd.tolist())  # Time must be provided in JD TDB
     try:
         result = query.vectors()
-    except Exception:  # Catch and re-raise all exceptions, and also provide query URL if generated
+    except Exception as e:  # Catch and re-raise all exceptions, and also provide query URL if generated
         if query.uri is not None:
             log.error(f"See the raw output from the JPL HORIZONS query at {query.uri}")
-        raise
+        raise e
+    finally:
+        query._session.close()
```

Comment:
👎 👎 👎 for using a private attribute from another package, is there another way to do this?

Reply:
I didn't have a close look at the astroquery docs but I couldn't see one. I could see if it could be done in a context manager?

Reply:
I can't see one from a quick poke around the astroquery source code.

Reply:
If the session isn't closing (or there's no public API to close the session) then this should also be reported upstream.

Reply:
it feels like the object should clean the session when it's deleted, but yeah opening an issue upstream is never a bad idea.

Reply:
I will open an issue. But I see what happens if I try it as a context.

Reply:
I opened an issue, but I have no issue with using this private function as it stands.

Reply:
🤷 if it's private for a reason so we probably shouldn't use it, and if there's no reason for it not to be private we can ask upstream to make it public. As long as someone opened an issue.